src/python/pants/core/goals/update_build_files.py

```diff
@@ -234,10 +234,17 @@ async def update_build_files(
         raise ValueError(f"Unrecognized formatter: {update_build_files_subsystem.formatter}")
 
     for request in union_membership[RewrittenBuildFileRequest]:
-        if update_build_files_subsystem.fmt and issubclass(request, chosen_formatter_request_class):
+        if update_build_files_subsystem.fmt and request == chosen_formatter_request_class:
```

Comment:
Note - it doesn't make sense to me that we use `issubclass` here, as `FormatWithRuffRequest`, `FormatWithBlackRequest` and `FormatWithYapfRequest` are all leaf classes and have no descendants. I think enforcing strict equality will also prevent nasty bugs in the future.

Reply:
That makes sense, can always chance in future if we need to.